tensorflow/core/kernels/data/map_dataset_op_test.cc

```diff
@@ -88,19 +58,28 @@ class MapDatasetOpTest : public DatasetOpsTestBase {
                                           {"output_types", map_output_dtypes},
                                           {"use_inter_op_parallelism", true},
                                           {"preserve_cardinality", false}});
-    std::unique_ptr<OpKernel> map_op_kernel;
-    TF_CHECK_OK(CreateOpKernel(map_node_def_, &map_op_kernel));
-    return map_op_kernel;
+    TF_CHECK_OK(CreateOpKernel(map_node_def_, map_kernel));
+    return Status::OK();
   }
 
- protected:
-  DatasetBase* range_dataset_;
-  DatasetBase* map_dataset_;
-  std::unique_ptr<OpKernel> map_kernel_;
-  std::unique_ptr<OpKernelContext> map_context_;
-  std::unique_ptr<IteratorContext> iterator_context_;
-  std::unique_ptr<IteratorBase> iterator_;
-  std::unique_ptr<SerializationContext> serialization_context_;
+  // Creates a new MapDataset op kernel context.
+  Status CreateMapDatasetContext(
+      DatasetBase* const input_dataset, OpKernel* const map_kernel,
+      std::unique_ptr<OpKernelContext>* map_context) {
+    map_inputs_.clear();
+    // Save the input dataset into a variant tensor as the input of MapDataset.
+    Tensor dataset_tensor(DT_VARIANT, TensorShape({}));
+    TF_RETURN_IF_ERROR(
+        StoreDatasetInVariantTensor(input_dataset, &dataset_tensor));
+    Variant variant = dataset_tensor.scalar<Variant>()();
+    TF_RETURN_IF_ERROR(AddDatasetInputFromArray<Variant>(
+        &map_inputs_, map_kernel->input_types(), TensorShape({}), {variant}));
+    input_dataset->Ref();
```

Comment:
Add `input_dataset->Ref();` to increase the reference count for `input_dataset`. Otherwise, will get the following error: `F ./tensorflow/core/lib/core/refcount.h:102] Check failed: ref_.load() > 0 (0 vs. 0)`.

IIUC, the reason is that `std::vector<std::unique_ptr<Tensor>> tensors_` used for garbage collecting also holds `input_dataset` inside a variant tensor.